variable/test/transform_test.cpp

```diff
@@ -43,35 +45,60 @@ auto make_variable_for_test(const Shape &shape, bool variances) {
   return var;
 }
 
-// TODO irregular partitions
-auto make_bin_indices(const scipp::index size, const scipp::index n_bins) {
-  std::vector<index_pair> aux(n_bins);
-  std::generate(begin(aux), end(aux),
+auto make_uniform_bin_indices(const scipp::index size,
+                              const scipp::index n_bins) {
+  std::vector<index_pair> index_pairs(n_bins);
+  std::generate(begin(index_pairs), end(index_pairs),
                 [lower = scipp::index{0}, size, n_bins]() mutable {
-                  const scipp::index upper = static_cast<scipp::index>(
-                      lower + (static_cast<double>(size) / n_bins));
+                  const auto upper = lower + static_cast<scipp::index>(
+                                                 (static_cast<double>(size) /
+                                                  static_cast<double>(n_bins)));
                   const index_pair res{lower, upper};
                   lower = upper;
                   return res;
                 });
-  aux.back().second = size;
-  return makeVariable<index_pair>(Dims{Dim::Event}, Shape{n_bins}, Values(aux));
+  index_pairs.back().second = size;
+  return makeVariable<index_pair>(Dims{Dim::Event}, Shape{n_bins},
+                                  Values(index_pairs));
+}
+
+auto make_random_bin_indices(const scipp::index size,
+                             const scipp::index n_bins) {
+  if (size == 0) {
+    return makeVariable<index_pair>(Dims{Dim::Event}, Shape{n_bins},
+                                    Values{index_pair{0, 0}});
+  }
+  Random rng;
+  const auto index_pairs =
+      index_pairs_from_sizes(rng.random_int_partition(size, n_bins));
+  return makeVariable<index_pair>(Dims{Dim::Event}, Shape{n_bins},
+                                  Values(index_pairs));
+}
+
+enum PartitionType { UNIFORM, RANDOM };
+
+auto make_bin_indices(const scipp::index size, const scipp::index n_bins,
+                      const PartitionType partition_type) {
+  return partition_type == PartitionType::UNIFORM
+             ? make_uniform_bin_indices(size, n_bins)
+             : make_random_bin_indices(size, n_bins);
```

Comment:
It feels quite intransparent what is happening here. Would we be better off explicitly spelling out a few examples/special cases?

Reply:
We might be. I am unhappy with how complicated this code is. But explicit test make it difficult to check large inputs and thereby our parallelism.

Reply:
But what aspect of our parallelism are we really testing here? It is impossible to see from the code.